tests/trace/test_weave_client.py

```diff
@@ -3569,3 +3569,489 @@ def test_sum_dict_leaves_deep_nested(client):
         },
         "top": [10, "mixed", 20],
     }
+
```

Comment:
2 things:
1. I am not seeing a test where you have a mix of rows (some of those rows have refs at the ref locations, where others just have data payloads). I am wondering if your query works in this case.
2. (less important) Would be nice to just have a new test file to isolate this component of the query layer (this test file is huge)